satellite/metainfo/metainfo.go

```diff
@@ -1724,7 +1726,7 @@ func (endpoint *Endpoint) BeginDeleteSegment(ctx context.Context, req *pb.Segmen
 
 	// moved from FinishDeleteSegment to avoid inconsistency if someone will not
 	// call FinishDeleteSegment on uplink side
-	err = endpoint.metainfo.Delete(ctx, path)
+	err = endpoint.metainfo.UnsynchronizedDelete(ctx, path)
```

Comment:
Should we have a synchronized delete here too?

What if one client tries to delete an object while another to re-upload it? Could it happen that the reuploaded pointers are deleted?

Reply:
We are inherently unsafe currently with concurrent calls to the database, unfortunately. This would require versions and other parts of metainfo index to fix.

Reply:
But if we could fail the delete with `ErrValueChanged`, this could safe the reuploaded pointer, right?

Reply:
The issue is that the "pointer bytes" should be from the point from when you call "BeginDeleteObject" not the beginning of this function call. So there's this time-gap we cannot fix right now. And that gap is much larger than what this change would fix here.

Reply:
Although, we might be able to use sat stream id to verify that the bytes match what's in the database to ensure some consistency. @mniewrzal thoughts?

Reply:
Ah, nevermind, it SatStreamID doesn't contain enough information to connect objects with segments.

Reply:
you want to add hash of pointerbytes into StreamID?

Reply:
I don't think we have a value that is shared between all segments (besides path), so I don't think there's a way at the moment.